### PR TITLE
Add test that ensure all files in source directory use `.d.ts` extension

### LINKED ISSUE #261: Meta: Add validation check to ensure all files in `source` directory use `.d.ts` extension
It's easy to forget the `.d` and make it `.ts`.

Should be a Node.js script that is run when testing: https://github.com/sindresorhus/type-fest/blob/b7e1c26c36f28a7a36f7f1e9d5fb7c07c438619a/package.json#L17

### PR DESCRIPTION
**Pull request by wafuwafu13**
Fixes #261

I want to make sure GitHub Actions fails as expected. https://github.com/wafuwafu13/type-fest/commit/c0358a821ac7298747a2ca23347543c89be0fdc7

### CODE REVIEW

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by wafuwafu13** [comment]
Thank you for reviewing!
I will be deleting these files after receive your review.
https://github.com/wafuwafu13/type-fest/commit/c0358a821ac7298747a2ca23347543c89be0fdc7

File: script/test/source-files-extension.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/* eslint-disable unicorn/prefer-module */
+const fs = require('fs');
+const process = require('process');
+
+const checkSourceFilesExtension = async () => {
+	try {
+		const files = await fs.promises.readdir('source');
+
+		let hasWrongExtensionFile = false;
```
Comment (sindresorhus) [inline script/test/source-files-extension.js:9]:
```suggestion
		let hasIncorrectFileExtension = false;
```

**Comment by sindresorhus** [comment]
Make sure you run `$ npm test` locally before pushing ;)

**Comment by wafuwafu13** [comment]
Done :sweat_drops:

**Comment by sindresorhus** [comment]
Thanks for contributing this :)

**Pull request merged**